rabbitmq/publisher.go

```diff
@@ -54,29 +53,65 @@ func NewRabbitMqPublisher(args ArgsRabbitMqPublisher) (*rabbitMqPublisher, error
 		cfg:                args.Config,
 		client:             args.Client,
 		closeChan:          make(chan struct{}),
-	}, nil
+	}
+
+	err = rp.checkAndCreateExchanges()
+	if err != nil {
+		return nil, err
+	}
+
+	return rp, nil
 }
 
 func checkArgs(args ArgsRabbitMqPublisher) error {
 	if check.IfNil(args.Client) {
 		return ErrNilRabbitMqClient
 	}
-	if args.Config.EventsExchange == "" {
-		return fmt.Errorf("empty events exchange name: %w", ErrInvalidRabbitMqExchangeName)
+
+	return nil
```

Comment:
please refactor: call in this function the checkAndCreateExchanges function (it doesn't need to be part of the rabbitMqPublisher struct).

Reply:
separated args check from exchanges creation